model/riscv_insts_vext_mem.sail

```diff
@@ -0,0 +1,1525 @@
+  VLE32     <-> 4,
+  VLE64     <-> 8
+}
+
+mapping vlewidth_pow : vlewidth <-> int = {
+  VLE8      <-> 3,
+  VLE16     <-> 4,
+  VLE32     <-> 5,
+  VLE64     <-> 6
+}
+
+mapping bytes_wordwidth : int <-> word_width = {
+  1 <-> BYTE,
+  2 <-> HALF,
+  4 <-> WORD,
+  8 <-> DOUBLE
+}
+
+/* Vector misaligned checking (used in vload) */
+val vcheck_misaligned : (xlenbits, word_width) -> bool effect {undef}
+function vcheck_misaligned(vaddr : xlenbits, width : word_width) -> bool = {
+  if   plat_enable_misaligned_access() then false
+  else match width {
+    BYTE     => false,
+    HALF     => vaddr[0] == bitone,
+    WORD     => vaddr[0] == bitone | vaddr[1] == bitone,
+    DOUBLE   => vaddr[0] == bitone | vaddr[1] == bitone | vaddr[2] == bitone
+  };
+}
+
+/* *****VLOAD**************VLETYPE(Vector Load Unit-Strided Normal, nf=0, mop=0, lumop=0)*************************** */
+union clause ast = VLETYPE : (bits(1), regidx, vlewidth, regidx)
+
+mapping clause encdec = VLETYPE(vm, rs1, width, vd)
+  <-> 0b000 @ 0b0 @ 0b00 @ vm @ 0b00000 @ rs1 @ encdec_vlewidth(width) @ vd @ 0b0000111
+
+val process_vle : forall 'b 'n 'p, (0 < 'b & 'b <= 8) & ('n >= 0). (bits(1), regidx, int('b), regidx, int('p), int('n)) -> Retired effect {escape, rmem, rmemt, rreg, undef, wmv, wmvt, wreg}
+function process_vle (vm, vd, load_width_bytes, rs1, EMUL_pow, num_elem) = {
+  let width_type : word_width = bytes_wordwidth(load_width_bytes);
+  let vm_val : vector('n, dec, bool) = read_vmask(num_elem, vm, vreg_name("v0"));
+  let vd_val : vector('n, dec, bits('b * 8)) = read_vreg(num_elem, load_width_bytes * 8, EMUL_pow, vd);
+  total : vector('n, dec, bits('b * 8)) = undefined;
+  mask_helper : vector('n, dec, bool) = undefined;
+  status : Retired = RETIRE_SUCCESS;
+
+  (total, mask_helper) = init_masked_result(num_elem, load_width_bytes * 8, EMUL_pow, vd_val, vm_val);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if mask_helper[i] == true then{
+      let elem_offset = i * load_width_bytes;
+      match ext_data_get_addr(rs1, to_bits(sizeof(xlen), elem_offset), Read(Data), width_type) {
+        Ext_DataAddr_Error(e)  => { ext_handle_data_check_error(e); status = RETIRE_FAIL },
+        Ext_DataAddr_OK(vaddr) => 
+          if vcheck_misaligned(vaddr, width_type)
+          then { handle_mem_exception(vaddr, E_Load_Addr_Align()); status = RETIRE_FAIL }
+          else match translateAddr(vaddr, Read(Data)) {
+            TR_Failure(e, _)     => { handle_mem_exception(vaddr, e); status = RETIRE_FAIL },
+            TR_Address(paddr, _) => {
+              match mem_read(Read(Data), paddr, load_width_bytes, false, false, false) {
+                MemValue(result) => total[i] = result,
+                MemException(e)  => { handle_mem_exception(vaddr, e); status = RETIRE_FAIL }
+              }
+            }
+          }
+      }
+    }
+  };
+
+  if status == RETIRE_SUCCESS then write_vreg(num_elem, load_width_bytes * 8, EMUL_pow, vd, total);
+  vstart = EXTZ(0b0);
```

Comment:
vector load and store instruction set the vstart when any kind of exception happens. How is this handled?

Reply:
The setting of vstart is added in the for loops of load / store execution functions in preparation for any exception.